From 1b2cf6e6c2640e08166af36eee775542a45325af Mon Sep 17 00:00:00 2001
From: "Yann E. MORIN" <yann.morin.1998@free.fr>
Date: Sun, 23 Oct 2016 22:36:40 +0200
Subject: [PATCH] build: get rid of sdkdir

Use of sdkdir causes problems during cross-compilation, where the full
path is then appended to the DESTDIR, leading to host paths being
appended in the target:
    https://bugs.busybox.net/show_bug.cgi?id=8696

Other drivers (e.g. keyboard) got rid of sdkdir. Do the same.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
---
 Makefile.am         | 2 --
 configure.ac        | 8 --------
 include/Makefile.am | 4 +++-
 xorg-libinput.pc.in | 5 +++--
 4 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index a447d61..7aa5523 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -19,8 +19,6 @@
 #  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 
-DISTCHECK_CONFIGURE_FLAGS = --with-sdkdir='$${includedir}/xorg'
-
 SUBDIRS = src include man test
 MAINTAINERCLEANFILES = ChangeLog INSTALL
 
diff --git a/configure.ac b/configure.ac
index 9894817..fe198cb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -63,14 +63,6 @@ AC_ARG_WITH(xorg-conf-dir,
             [xorgconfdir='${prefix}/share/X11/xorg.conf.d'])
 AC_SUBST(xorgconfdir)
 
-# X Server SDK location is required to install header files
-sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
-
-# Workaround overriding sdkdir to be able to create a tarball when user has no
-# write permission in sdkdir. See DISTCHECK_CONFIGURE_FLAGS in Makefile.am
-AC_ARG_WITH([sdkdir], [], [sdkdir="$withval"])
-AC_SUBST([sdkdir])
-
 DRIVER_NAME=libinput
 AC_SUBST([DRIVER_NAME])
 
diff --git a/include/Makefile.am b/include/Makefile.am
index 0d52b9b..271ffac 100644
--- a/include/Makefile.am
+++ b/include/Makefile.am
@@ -1 +1,3 @@
-sdk_HEADERS = libinput-properties.h
+# Location formerly known as 'sdkdir'
+xorgincludedir = $(includedir)/xorg
+xorginclude_HEADERS = libinput-properties.h
diff --git a/xorg-libinput.pc.in b/xorg-libinput.pc.in
index ff98189..9c0f506 100644
--- a/xorg-libinput.pc.in
+++ b/xorg-libinput.pc.in
@@ -1,6 +1,7 @@
-sdkdir=@sdkdir@
+prefix=@prefix@
+includedir=@includedir@
 
 Name: xorg-libinput
 Description: X.Org libinput input driver.
 Version: @PACKAGE_VERSION@
-Cflags: -I${sdkdir}
+Cflags: -I${includedir}/xorg
-- 
2.7.4

